rdkit/Chem/Draw/cairoCanvas.py

```diff
@@ -10,44 +10,120 @@
   except ImportError:
     import cairo
+  else:
+    have_cairocffi = True
 else:
   try:
     import cairo
   except ImportError:
-    import cairocffi as cairo
+    try:
+      import cairocffi as cairo
+    except:
+      raise
+    else:
+      have_cairocffi = True
+have_pango = False
+if not 'RDK_NOPANGO' in os.environ:
+  if (have_cairocffi):
+    import cffi
+    import platform
+    ffi = cffi.FFI()
+    ffi.cdef('''
+        /* GLib */
+        typedef void* gpointer;
+        typedef void cairo_t;
+        typedef void PangoFontDescription;
+        void g_object_unref (gpointer object);
 
+        /* Pango and PangoCairo */
+        #define PANGO_SCALE 1024
+        typedef ... PangoLayout;
+        typedef enum {
+            PANGO_ALIGN_LEFT,
+            PANGO_ALIGN_CENTER,
+            PANGO_ALIGN_RIGHT
+        } PangoAlignment;
+        typedef struct PangoRectangle {
+          int x;
+          int y;
+          int width;
+          int height;
+        } PangoRectangle;
+        PangoLayout *pango_cairo_create_layout (cairo_t *cr);
+        void pango_cairo_update_layout (cairo_t *cr, PangoLayout *layout);
+        void pango_cairo_show_layout (cairo_t *cr, PangoLayout *layout);
+        void pango_layout_set_alignment (
+            PangoLayout *layout, PangoAlignment alignment);
+        void pango_layout_set_markup (
+            PangoLayout *layout, const char *text, int length);
+        void pango_layout_get_pixel_extents (PangoLayout *layout,
+            PangoRectangle *ink_rect, PangoRectangle *logical_rect);
+        PangoFontDescription *pango_font_description_new (void);
+        void pango_font_description_free (PangoFontDescription *desc);
+        void pango_font_description_set_family (PangoFontDescription *desc,
+            const char *family);
+        void pango_font_description_set_size (PangoFontDescription *desc,
+            int size);
+        void pango_layout_set_font_description (PangoLayout *layout,
+            const PangoFontDescription *desc);
+    ''')
+    if (platform.system() == 'Windows'):
+      defaultLibs = {
+        'pango_default_lib': 'libpango-1.0-0.dll',
+        'pangocairo_default_lib': 'libpangocairo-1.0-0.dll',
+        'gobject_default_lib': 'libgobject-2.0-0.dll'
+      }
+    else:
+      defaultLibs = {
+        'pango_default_lib': 'pango-1.0',
+        'pangocairo_default_lib': 'pangocairo-1.0',
+        'gobject_default_lib': 'gobject-2.0'
+      }
+    import ctypes.util
+    for libType in ['pango', 'pangocairo', 'gobject']:
+      envVar = 'RDK_' + libType.upper() + '_LIB'
+      if (envVar in os.environ):
+        libName = os.environ[envVar]
+      else:
+        libName = defaultLibs[libType + '_default_lib']
+      libPath = ctypes.util.find_library(libName)
+      exec(libType + ' = None')
+      if (libPath):
+        try:
+          exec(libType + ' = ffi.dlopen("' \
+            + libPath.replace('\\', '\\\\') + '")')
+        except:
+          pass
```

Comment:
If the RDK_ environments are set and the libs aren't found, I think it it reasonable to raise an exception along the lines of

```
ImportError: RDK_PANGO_LIB set to /usr/lib/libpango..., but PANGO library cannot be loaded.
```